From: Julia Lawall <julia@diku.dk>
Date: Sat, 13 Mar 2010 12:22:16 -0800
Subject: [PATCH] drivers/net/tg3.c: change the field used with the
 TG3_FLAG_10_100_ONLY constant
References: bnc#616088
Git-commit: 3f07d1295191cfa41125e4e61ee2064790070071
Patch-mainline: v2.6.34-rc2

The constant TG3_FLAG_10_100_ONLY should be used with the tg3_flags field,
not the tg3_flags2 field, as done elsewhere in the same file.

Signed-off-by: Julia Lawall <julia@diku.dk>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Brandon Philips <bphilips@suse.de>

---
 drivers/net/tg3.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/net/tg3.c b/drivers/net/tg3.c
index c3b4fe7..22cf1c4 100644
--- a/drivers/net/tg3.c
+++ b/drivers/net/tg3.c
@@ -9776,7 +9776,7 @@ static int tg3_set_settings(struct net_device *dev, struct ethtool_cmd *cmd)
 			   ADVERTISED_Pause |
 			   ADVERTISED_Asym_Pause;
 
-		if (!(tp->tg3_flags2 & TG3_FLAG_10_100_ONLY))
+		if (!(tp->tg3_flags & TG3_FLAG_10_100_ONLY))
 			mask |= ADVERTISED_1000baseT_Half |
 				ADVERTISED_1000baseT_Full;
 
-- 
1.7.1

